File: app/src/main/java/party/qwer/Iris/ImageDeleter.java
package party.qwer.Iris;

import java.io.File;
import java.util.concurrent.Executors;
import java.util.concurrent.ScheduledExecutorService;
import java.util.concurrent.TimeUnit;

public class ImageDeleter {
    private final String imageDirPath;
    private final long deletionInterval;
    private volatile boolean running = true;
    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();

    public ImageDeleter(String imageDirPath, long deletionInterval) {
        this.imageDirPath = imageDirPath;
        this.deletionInterval = deletionInterval;
    }

    public void startDeletion() {
        scheduler.scheduleWithFixedDelay(this::deleteOldImages, 0, deletionInterval, TimeUnit.MILLISECONDS);
    }

    public void stopDeletion() {
        running = false;
        scheduler.shutdown();
        try {
            if (!scheduler.awaitTermination(5, TimeUnit.SECONDS)) {
                scheduler.shutdownNow();
            }
        } catch (InterruptedException e) {
            scheduler.shutdownNow();
            Thread.currentThread().interrupt();
            System.err.println("Error shutting down image deletion scheduler: " + e);
        }
    }

    private void deleteOldImages() {
        if (!running) {
            System.out.println("Image deletion task stopped.");
            scheduler.shutdown();
            return;
        }

        File imageDir = new File(imageDirPath);
        if (!imageDir.exists() || !imageDir.isDirectory()) {
            System.out.println("Image directory does not exist: " + imageDirPath);
            return;
        }

        long oneDayAgo = System.currentTimeMillis() - TimeUnit.DAYS.toMillis(1);
        File[] files = imageDir.listFiles();
        if (files != null) {
            for (File file : files) {
                if (file.isFile() && file.lastModified() < oneDayAgo) {
                    if (file.delete()) {
                        System.out.println("Deleted old image file: " + file.getName());
                    } else {
                        System.err.println("Failed to delete image file: " + file.getName());
                    }
                }
            }
        }
    }
}

File: app/src/main/java/party/qwer/Iris/ImageDeleter.kt
package party.qwer.Iris

import java.io.File
import java.util.concurrent.TimeUnit

class ImageDeleter(private val imageDirPath: String, private val deletionInterval: Long) {
    private var running = true

    fun startDeletion() {
        Thread {
            while (running) {
                deleteOldImages()
                try {
                    Thread.sleep(deletionInterval)
                } catch (e: InterruptedException) {
                    System.err.println("Image deletion thread interrupted: $e")
                    running = false
                    break
                }
            }
        }.start()
    }

    fun stopDeletion() {
        running = false
    }

    private fun deleteOldImages() {
        val imageDir = File(imageDirPath)
        if (!imageDir.exists() || !imageDir.isDirectory) {
            println("Image directory does not exist: $imageDirPath")
            return
        }

        val oneDayAgo = System.currentTimeMillis() - TimeUnit.DAYS.toMillis(1)
        val files = imageDir.listFiles()
        if (files != null) {
            for (file in files) {
                if (file.isFile && file.lastModified() < oneDayAgo) {
                    if (file.delete()) {
                        println("Deleted old image file: " + file.name)
                    } else {
                        System.err.println("Failed to delete image file: " + file.name)
                    }
                }
            }
        }
    }
}
